src/components/molecules/NavBar/components/NavBarRadio/NavBarRadio.tsx

```diff
@@ -0,0 +1,124 @@
+
+import "./NavBarRadio.scss";
+
+export const NavBarRadio: React.FC = () => {
+  const venue = useSelector(currentVenueSelectorData);
+
+  const radioStations = useSelector(radioStationsSelector) ?? [];
+  const [isRadioPlaying, setIsRadioPlaying] = useState(false);
+  const hasRadioStations = radioStations && radioStations.length;
+  const radioStation = !!hasRadioStations && radioStations[0];
+  const isSoundCloud =
+    !!hasRadioStations && RegExp("soundcloud").test(radioStations[0]);
+
+  const sound = useMemo(
+    () =>
+      radioStations && hasElements(radioStations) && !isSoundCloud
+        ? new Audio(radioStations[0])
+        : undefined,
+    [isSoundCloud, radioStations]
+  );
+  const handleRadioEnable = useCallback(() => setIsRadioPlaying(true), []);
+
+  const [showRadioPopover, setShowRadioPopover] = useState(false);
+
+  const toggleShowRadioPopover = useCallback(
+    () => setShowRadioPopover((prevState) => !prevState),
+    []
+  );
+  const { volume, setVolume } = useRadio(isRadioPlaying, sound);
+  const showNormalRadio = (venue?.showRadio && !isSoundCloud) ?? false;
+  const showSoundCloudRadio = (venue?.showRadio && isSoundCloud) ?? false;
+
+  const radioFirstPlayStateLoaded = useRef(false);
+  const showRadioOverlay = useMemo(() => {
+    if (!radioFirstPlayStateLoaded.current) {
+      const radioFirstPlayStorageKey = "radioFirstPlay";
+      const radioFirstPlayState = localStorage.getItem(
+        radioFirstPlayStorageKey
+      );
+      if (!radioFirstPlayState) {
+        localStorage.setItem(radioFirstPlayStorageKey, JSON.stringify(true));
+        return true;
+      }
+      radioFirstPlayStateLoaded.current = true;
+    }
+    return false;
+  }, [radioFirstPlayStateLoaded]);
+
+  if (showNormalRadio) {
+    return (
+      <OverlayTrigger
+        trigger="click"
+        placement="bottom-end"
+        rootClose={true}
+        defaultShow={showRadioOverlay}
+        overlay={
+          <Popover id="radio-popover">
+            <Popover.Content>
+              <RadioModal
+                {...{
+                  volume,
+                  setVolume,
+                  title: venue?.radioTitle,
+                }}
+                onEnableHandler={handleRadioEnable}
+                isRadioPlaying={isRadioPlaying}
+              />
+            </Popover.Content>
+          </Popover>
+        }
+      >
+        <div
+          className={classNames("profile-icon navbar-link-radio", {
+            off: volume === 0,
+          })}
+        />
+      </OverlayTrigger>
+    );
+  }
+
+  const soundCloudIframeSrc = `https://w.soundcloud.com/player/?url=${radioStation}&amp;start_track=0&amp;single_active=true&amp;show_artwork=false`;
+
+  const radioTriggerClasses = classNames("radio-trigger__iframe", {
+    "radio-trigger__iframe--visible": showRadioPopover,
+    "radio-trigger__iframe--hidden": !showRadioPopover,
+  });
+
+  return (
+    <>
+      {showSoundCloudRadio && (
```

Comment:
Instead of wrapping this whole block in a fragment + check like this, we can just have a 'guard' line before this JSX block:

```suggestion
  if (!showSoundCloudRadio) return null;
  
  return (
```